sync/plocker_test.go

```diff
@@ -45,16 +46,16 @@ func TestPartitionLocker(t *testing.T) {
 		}
 		var s l
 		s.locker = *sync.NewPartitionLocker()
-		var locks int
+		var locks atomic.Int64
 		const id = "id"
 		s.locker.Lock(id)
 		go func() {
 			s.locker.Lock(id)
-			locks = locks + 1
+			locks.Store(locks.Add(1))
```

Comment:
`Add` already adds, no need to store the returned value.

```suggestion
			locks.Add(1)
```

Anyway, why using another sync mechanism? Isn't the partition locker enough to avoid a race condition here?